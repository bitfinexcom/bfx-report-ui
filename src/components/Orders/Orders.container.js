import { connect } from 'react-redux'
import actions from 'state/orders/actions'
import Orders from './Orders'

function mapStateToProps(state = {}) {
  return {
    entries: state.orders.entries,
    loading: !state.orders.dataReceived,
  }
}

const mapDispatchToProps = dispatch => ({
  fetchOrders: () => dispatch(actions.fetchOrders()),
})

const OrdersContainer = connect(mapStateToProps, mapDispatchToProps)(Orders)

export default OrdersContainer
