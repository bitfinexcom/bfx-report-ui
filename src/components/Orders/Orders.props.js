import PropTypes from 'prop-types'
import { intlShape } from 'react-intl'

const ORDERS_ENTRIES_PROPS = PropTypes.shape({
  id: PropTypes.number.isRequired,
  pair: PropTypes.string.isRequired,
  mtsUpdate: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  priceAvg: PropTypes.number.isRequired,
})

export const propTypes = {
  entries: PropTypes.arrayOf(ORDERS_ENTRIES_PROPS).isRequired,
  intl: intlShape.isRequired,
  loading: PropTypes.bool.isRequired,
}

export const defaultProps = {
  entries: [],
  intl: {},
  loading: true,
}
