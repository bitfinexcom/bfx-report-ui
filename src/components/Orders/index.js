export { default } from './Orders.container'
