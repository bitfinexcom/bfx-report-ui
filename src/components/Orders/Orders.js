import React, { Fragment, PureComponent } from 'react'
import { injectIntl } from 'react-intl'
import {
  Button,
  Card,
  Elevation,
} from '@blueprintjs/core'
import {
  Cell,
  Column,
  Table,
  TruncatedFormat,
} from '@blueprintjs/table'
import Loading from 'components/Loading'
import NoData from 'components/NoData'
import { formatTime } from 'state/utils'
import { propTypes, defaultProps } from './Orders.props'

const COLUMN_WIDTHS = [80, 70, 150, 100, 100, 100, 100, 150, 200]

class Orders extends PureComponent {
  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.fetchOrders()
  }

  componentDidUpdate(prevProps) {
    const { loading, fetchOrders } = this.props
    const { prevLoading } = prevProps
    if (loading && loading !== prevLoading) {
      fetchOrders()
    }
  }

  render() {
    const {
      entries,
      intl,
      loading,
    } = this.props
    const numRows = entries.length

    const idCellRenderer = rowIndex => (
      <Cell wrapText={false}>
        {entries[rowIndex].id}
      </Cell>
    )

    const pairCellRenderer = rowIndex => (
      <Cell>
        {entries[rowIndex].pair}
      </Cell>
    )

    const typeCellRenderer = rowIndex => (
      <Cell>
        {entries[rowIndex].type}
      </Cell>
    )

    const amountOrigCellRenderer = rowIndex => (
      <Cell>
        {entries[rowIndex].amountOrig}
      </Cell>
    )

    const amountCellRenderer = rowIndex => (
      <Cell>
        {entries[rowIndex].amount}
      </Cell>
    )

    const priceCellRenderer = rowIndex => (
      <Cell>
        {entries[rowIndex].price}
      </Cell>
    )

    const priceAvgCellRenderer = rowIndex => (
      <Cell>
        {entries[rowIndex].priceAvg}
      </Cell>
    )

    const mtsUpdateCellRenderer = rowIndex => (
      <Cell>
        <TruncatedFormat>
          {formatTime(entries[rowIndex].mtsUpdate)}
        </TruncatedFormat>
      </Cell>
    )

    const statusCellRenderer = rowIndex => (
      <Cell>
        {entries[rowIndex].status}
      </Cell>
    )

    let showContent
    if (loading) {
      showContent = (
        <Loading title='orders.title' />
      )
    } else if (numRows === 0) {
      showContent = (
        <NoData title='orders.title' />
      )
    } else {
      showContent = (
        <Fragment>
          <h4>
            {intl.formatMessage({ id: 'orders.title' })}
            &nbsp;
            <Button icon='cloud-download' disabled>
              {intl.formatMessage({ id: 'timeframe.download' })}
            </Button>
          </h4>
          <Table
            className='bitfinex-table'
            numRows={numRows}
            enableRowHeader={false}
            columnWidths={COLUMN_WIDTHS}
          >
            <Column
              id='id'
              name='#'
              cellRenderer={idCellRenderer}
            />
            <Column
              id='symbol'
              name={intl.formatMessage({ id: 'orders.column.pair' })}
              cellRenderer={pairCellRenderer}
            />
            <Column
              id='type'
              name={intl.formatMessage({ id: 'orders.column.type' })}
              cellRenderer={typeCellRenderer}
            />
            <Column
              id='amount'
              name={intl.formatMessage({ id: 'orders.column.amount' })}
              cellRenderer={amountCellRenderer}
            />
            <Column
              id='amountOrig'
              name={intl.formatMessage({ id: 'orders.column.amount-orig' })}
              cellRenderer={amountOrigCellRenderer}
            />
            <Column
              id='price'
              name={intl.formatMessage({ id: 'orders.column.price' })}
              cellRenderer={priceCellRenderer}
            />
            <Column
              id='priceAvg'
              name={intl.formatMessage({ id: 'orders.column.avgprice' })}
              cellRenderer={priceAvgCellRenderer}
            />
            <Column
              id='mtsUpdate'
              name={intl.formatMessage({ id: 'orders.column.update' })}
              cellRenderer={mtsUpdateCellRenderer}
            />
            <Column
              id='status'
              name={intl.formatMessage({ id: 'orders.column.status' })}
              cellRenderer={statusCellRenderer}
            />
          </Table>
        </Fragment>
      )
    }

    return (
      <Card interactive elevation={Elevation.ZERO} className='col-lg-12 col-md-12 col-sm-12 col-xs-12'>
        {showContent}
      </Card>
    )
  }
}

Orders.propTypes = propTypes
Orders.defaultProps = defaultProps

export default injectIntl(Orders)
