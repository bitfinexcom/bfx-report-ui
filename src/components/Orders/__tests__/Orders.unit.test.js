import React from 'react'
import ReactDOM from 'react-dom'
import {
  renderWithIntl,
  rendererCreateWithIntl,
} from 'tests/helper'
import Orders from '../Orders'

const NO_ENTRY = []
const ENTRIES = [
  {
    id: 1149732382,
    gid: null,
    cid: 53916927769,
    pair: 'BTC/USD',
    mtsCreate: 1529938717000,
    mtsUpdate: 1529941979000,
    amount: 0,
    amountOrig: -0.2,
    type: 'EXCHANGE LIMIT',
    typePrev: null,
    flags: '0',
    status: 'EXECUTED @ 18037.0(-0.2)',
    price: 18037,
    priceAvg: 18037,
    priceTrailing: 0,
    priceAuxLimit: 0,
    notify: 0,
    placedId: null,
  },
  {
    id: 1149721794,
    gid: null,
    cid: 10241082294,
    pair: 'BTC/USD',
    mtsCreate: 1529549441000,
    mtsUpdate: 1529576031000,
    amount: 0,
    amountOrig: 0.2,
    type: 'EXCHANGE LIMIT',
    typePrev: null,
    flags: '0',
    status: 'EXECUTED @ 16560.0(0.2)',
    price: 16560,
    priceAvg: 16560,
    priceTrailing: 0,
    priceAuxLimit: 0,
    notify: 0,
    placedId: null,
  },
  {
    id: 1149721793,
    gid: null,
    cid: 10230273945,
    pair: 'BTC/USD',
    mtsCreate: 1529549430000,
    mtsUpdate: 1529575757000,
    amount: 0,
    amountOrig: -0.5,
    type: 'EXCHANGE LIMIT',
    typePrev: null,
    flags: '0',
    status: 'EXECUTED @ 18061.0(-0.5)',
    price: 18061,
    priceAvg: 18061,
    priceTrailing: 0,
    priceAuxLimit: 0,
    notify: 0,
    placedId: null,
  },
  {
    id: 1149721791,
    gid: null,
    cid: 10212323001,
    pair: 'BTC/USD',
    mtsCreate: 1529549412000,
    mtsUpdate: 1529575721000,
    amount: 0,
    amountOrig: 0.4,
    type: 'EXCHANGE LIMIT',
    typePrev: null,
    flags: '0',
    status: 'EXECUTED @ 17410.0(0.4)',
    price: 17410,
    priceAvg: 17410,
    priceTrailing: 0,
    priceAuxLimit: 0,
    notify: 0,
    placedId: null,
  },
  {
    id: 1149717540,
    gid: null,
    cid: 23759415063,
    pair: 'BTC/USD',
    mtsCreate: 1529476559000,
    mtsUpdate: 1529549557000,
    amount: 0.2,
    amountOrig: 0.2,
    type: 'EXCHANGE LIMIT',
    typePrev: null,
    flags: '0',
    status: 'CANCELED',
    price: 17410,
    priceAvg: 0,
    priceTrailing: 0,
    priceAuxLimit: 0,
    notify: 0,
    placedId: null,
  },
  {
    id: 1149721792,
    gid: null,
    cid: 10219626687,
    pair: 'BTC/USD',
    mtsCreate: 1529549420000,
    mtsUpdate: 1529549554000,
    amount: -0.07400001,
    amountOrig: -0.07400001,
    type: 'EXCHANGE LIMIT',
    typePrev: null,
    flags: '0',
    status: 'CANCELED',
    price: 18042,
    priceAvg: 0,
    priceTrailing: 0,
    priceAuxLimit: 0,
    notify: 0,
    placedId: null,
  },
  {
    id: 1149717543,
    gid: null,
    cid: 31589172329,
    pair: 'BTC/USD',
    mtsCreate: 1529484389000,
    mtsUpdate: 1529549265000,
    amount: -0.003,
    amountOrig: -0.023,
    type: 'EXCHANGE LIMIT',
    typePrev: null,
    flags: '0',
    status: 'CANCELED was: PARTIALLY FILLED @ 18030.0(-0.02)',
    price: 18030,
    priceAvg: 18030,
    priceTrailing: 0,
    priceAuxLimit: 0,
    notify: 0,
    placedId: null,
  },
  {
    id: 1149717527,
    gid: null,
    cid: 43119658499,
    pair: 'BTC/USD',
    mtsCreate: 1529409520000,
    mtsUpdate: 1529549259000,
    amount: 1,
    amountOrig: 1,
    type: 'EXCHANGE LIMIT',
    typePrev: null,
    flags: '0',
    status: 'CANCELED',
    price: 17.6,
    priceAvg: 0,
    priceTrailing: 0,
    priceAuxLimit: 0,
    notify: 0,
    placedId: null,
  },
  {
    id: 1149715964,
    gid: null,
    cid: 12175783466,
    pair: 'BTC/USD',
    mtsCreate: 1527564176000,
    mtsUpdate: 1527625559000,
    amount: 0,
    amountOrig: -1,
    type: 'EXCHANGE LIMIT',
    typePrev: null,
    flags: 0,
    status: 'EXECUTED @ 1020.0(-0.81): was PARTIALLY FILLED @ 1020.0(-0.09), PARTIALLY FILLED @ 1020.0(-0.1)',
    price: 1020,
    priceAvg: 1020,
    priceTrailing: 0,
    priceAuxLimit: 0,
    notify: 0,
    placedId: null,
  },
]

test('renders without crashing', () => {
  const div = document.createElement('div')
  renderWithIntl(<Orders />, div)
  ReactDOM.unmountComponentAtNode(div)
})

test('renders without crashing with data', () => {
  const div = document.createElement('div')
  renderWithIntl(<Orders loading={false} entries={ENTRIES} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

test('render loading correctly', () => {
  const component = rendererCreateWithIntl(<Orders />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('render no data correctly', () => {
  const component = rendererCreateWithIntl(<Orders loading={false} entries={NO_ENTRY} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// test('render correctly with data', () => {
//   const component = rendererCreateWithIntl(<Orders loading={false} entries={ENTRIES} />)
//   const tree = component.toJSON()
//   expect(tree).toMatchSnapshot()
// })
