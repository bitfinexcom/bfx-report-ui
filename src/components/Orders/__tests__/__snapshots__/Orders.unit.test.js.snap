// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render loading correctly 1`] = `
<div
  className="pt-card pt-interactive pt-elevation-0 col-lg-12 col-md-12 col-sm-12 col-xs-12"
>
  <h4>
    Orders
  </h4>
  <div>
    Loading...
  </div>
</div>
`;

exports[`render no data correctly 1`] = `
<div
  className="pt-card pt-interactive pt-elevation-0 col-lg-12 col-md-12 col-sm-12 col-xs-12"
>
  <h4>
    Orders
  </h4>
  <div>
    No related data in this time range. You can try another time range.
  </div>
</div>
`;
